src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -464,6 +464,11 @@ def config_source_control(resource_group_name, name, repo_url, repository_type=N
             sc = SourceControl(location, name='GitHub', token=git_token)
             client.update_source_control('GitHub', sc)
 
+        # service doesn't support update for being a less likely scenario, but no error is emitted.
+        # we work around it by deleting the existing binding.
+        if show_source_control(resource_group_name, name, slot).repo_url:
+            delete_source_control(resource_group_name, name, slot)
```

Comment:
Only repo-url argument is required, all other args have reasonable defaults.
I believe they support update style, but with the issue being worked around, I am not so sure now :).